src/tests/unit/tests/injected/element-based-view-model-creator.test.ts

```diff
@@ -33,59 +38,60 @@ describe('ElementBasedViewModelCreator', () => {
-                { results: [] } as UnifiedScanResultStoreData,
-                {} as CardSelectionStoreData,
-            ),
-        ).toBeNull();
+    afterEach(() => {
+        convertStoreDataForScanNodeResultsCallbackMock.verifyAll();
     });
 
     test('getElementBasedViewModel: results are null', () => {
-        expect(
-            testSubject.getElementBasedViewModel(
-                { rules: [] } as UnifiedScanResultStoreData,
-                {} as CardSelectionStoreData,
-            ),
-        ).toBeNull();
+        const dataStub = {} as UnifiedScanResultStoreData;
+        const cardSelectionDataStub = {} as CardSelectionStoreData;
+        convertStoreDataForScanNodeResultsCallbackMock
+            .setup(mock => mock(dataStub, cardSelectionDataStub))
+            .returns(() => null)
+            .verifiable(Times.once());
+        expect(testSubject.getElementBasedViewModel(dataStub, cardSelectionDataStub)).toBeNull();
     });
 
     test('getElementBasedViewModel: cardSelectionData are null', () => {
-        expect(
-            testSubject.getElementBasedViewModel(
-                { rules: [], results: [] } as UnifiedScanResultStoreData,
-                null,
-            ),
-        ).toBeNull();
+        const dataStub = {} as UnifiedScanResultStoreData;
+        const cardSelectionDataStub = null;
+        convertStoreDataForScanNodeResultsCallbackMock
+            .setup(mock => mock(dataStub, cardSelectionDataStub))
+            .returns(() => [])
+            .verifiable(Times.once());
+        expect(testSubject.getElementBasedViewModel(dataStub, cardSelectionDataStub)).toBeNull();
     });
 
     test('getElementBasedViewModel: cardSelectionData.rules are null', () => {
-        expect(
-            testSubject.getElementBasedViewModel(
-                { rules: [], results: [] } as UnifiedScanResultStoreData,
-                { rules: null, visualHelperEnabled: true, focusedResultUid: null },
-            ),
-        ).toBeNull();
+        const dataStub = {} as UnifiedScanResultStoreData;
+        const cardSelectionDataStub = {
+            rules: null,
+            visualHelperEnabled: true,
+            focusedResultUid: null,
+        };
+        convertStoreDataForScanNodeResultsCallbackMock
+            .setup(mock => mock(dataStub, cardSelectionDataStub))
+            .verifiable(Times.once());
+        expect(testSubject.getElementBasedViewModel(dataStub, cardSelectionDataStub)).toBeNull();
     });
 
     test('getElementBasedViewModel: no highlighted results', () => {
         const unifiedResult = exampleUnifiedResult;
-        const scanResultStoreData = {
-            results: [unifiedResult],
-            rules: [],
-            platformInfo: null,
-        } as UnifiedScanResultStoreData;
+        const ruleStub = { id: unifiedResult.ruleId } as UnifiedRule;
+        const ScanNodeResults: ScanNodeResult[] = [{ ...unifiedResult, rule: ruleStub }];
```

Comment:
nit: casing on the variable
```suggestion
        const scanNodeResults: ScanNodeResult[] = [{ ...unifiedResult, rule: ruleStub }];
```